lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -441,7 +441,10 @@ ParseResult FirRegOp::parse(OpAsmParser &parser, OperationState &result) {
     OptionalParseResult presetIntResult =
         parser.parseOptionalInteger(presetValue.emplace());
     if (!presetIntResult.has_value() || failed(*presetIntResult))
-      return parser.emitError(loc, "expected integer value");
+      return parser.emitError(presetValueLoc, "expected integer value");
+    if (presetValue->isNegative())
+      return parser.emitError(presetValueLoc,
+                              "preset value must not be negative");
```

Comment:
Thanks for correcting the location. I assume this will get the carat indicator in the correct place now?

Reply:
```
error: custom op 'seq.firreg' expected integer value
    %preset_0 = seq.firreg %next32 clock %clock preset NaN : i32
                                                       ^
```